h/static/scripts/group-forms/hooks/use-update-moderation-status.ts

```diff
@@ -0,0 +1,67 @@
+import {
+  useCallback,
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from 'preact/hooks';
+
+import { Config } from '../config';
+import type { APIAnnotationData, ModerationStatus } from '../utils/api';
+import { callAPI } from '../utils/api';
+
+/**
+ * Returns a callback that can be used to update the moderation status of an
+ * annotation via API call
+ */
+export function useUpdateModerationStatus(annotation: APIAnnotationData) {
+  const config = useContext(Config);
+
+  // Track the last in-flight request
+  const abortCtrlRef = useRef<AbortController>();
+  useEffect(() => {
+    const abortCtrl = abortCtrlRef.current;
+    return () => abortCtrl?.abort();
+  }, []);
+
+  const [updating, setUpdating] = useState(false);
+  const updateModerationStatus = useCallback(
+    async (moderationStatus: ModerationStatus) => {
+      const annotationId = annotation.id;
+      if (!config?.api.annotationModeration || !annotationId) {
+        return;
+      }
+
+      const { url, ...apiConfig } = config.api.annotationModeration;
+
+      if (abortCtrlRef.current) {
+        abortCtrlRef.current.abort();
+      }
+      abortCtrlRef.current = new AbortController();
+
+      setUpdating(true);
+      try {
+        // TODO Handle errors
```

Comment:
What is the plan for handling errors? Aside from conflicting changes as mentioned on the PR, I mean also more mundane issues like a network error.

Reply:
Actually scratch that. That is probably not the pattern we want to follow here, because the error would represent the error from the last request, so that could be confusing.

I'll check how we do this in the LMS and try to implement this similarly.